### PR TITLE
Cleaning up nodes

### PR DESCRIPTION
**Pull request by nscarcella**
This is a rather large pile of shit, and I know I have to make it work with the linker, but I wanted to check the changes are cool.

### CODE REVIEW

File: tests/interpreter.test.js [hunk 1]
```diff
@@ -122,13 +114,12 @@ const fixture = new Map([
   //-------------------------------------------------------------------------------------------------------------------------------
 
   [Literal(null), null],
-  [SelfLiteral, this],
+  [Variable('this'), this],
```
Comment (javierfernandes) [inline tests/interpreter.test.js]: was this on purpose ?
Wollok keyword is `self` (although I didn't like it :P it was changed from this to self a while ago)

**Comment by nscarcella** [comment]
@javierfernandes I merged the bulk of the change, but can't seem to be able to make the linker tests pass. (I have no idea what the linker does anyway :P )
Any hints are welcome

P.S.: What does the ::this thingie do?

**Comment by javierfernandes** [comment]
Ok, I'm looking at it no problem.

Seems like you deleted "model/visiting.js" which was not a model but
utilities to visit :P

::object.method

Is the bind operator.
Binds the "this" so you can avoid doing

object.method.bind(object)

**Comment by nscarcella** [comment]
I moved it to the linker folder.

**Comment by javierfernandes** [comment]
Hehe.. yes VSCode doesn't support it in the editor. Sucks.
If you find a way to make it "understand it" then let me know. I just ignore that one :S

Ok. I've fixed the linker now. 
I'll merge this

**Pull request merged**